standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java

```diff
@@ -956,14 +986,16 @@ private TTransport createAuthBinaryTransport(URI store, TTransport underlyingTra
         // tokenSig could be null
         tokenStrForm = SecurityUtils.getTokenStrForm(tokenSig);
 
-        if (tokenStrForm != null) {
+        if (tokenStrForm != null && !fallbackToKerberos) {
```

Comment:
Earlier tokenStrForm is null, it will fallback to kerberos without fallbackToKerberos token